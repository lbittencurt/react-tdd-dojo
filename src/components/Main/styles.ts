import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #fafafb;
  color: #4a4a4a;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`

export const Title = styled.h1`
  color: #4a4a4a;
  font-size: 2.5rem;
`

export const Description = styled.h2`
  color: #797979;
  font-size: 2rem;
  font-weight: 400;
`

export const Illustration = styled.img`
  margin-bottom: 3rem;
  width: min(10rem, 100%);
`
