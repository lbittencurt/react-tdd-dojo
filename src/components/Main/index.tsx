import * as S from './styles'

const Main = ({
  title = 'React Boilerplate',
  description = 'TypeScript, ReactJS, NextJS e Styled Components'
}) => (
  <S.Wrapper>
    <S.Illustration
      src="/img/react-atom-icon.svg"
      alt="Um desenvolvedor de frente para uma tela com cÃ³digo."
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
  </S.Wrapper>
)

export default Main
