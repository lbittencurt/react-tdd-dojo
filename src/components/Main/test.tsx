import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react boilerplate/i })
    ).toBeInTheDocument()
  })

  it('should render component and validate snapshot', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#fafafb' })
  })
})
