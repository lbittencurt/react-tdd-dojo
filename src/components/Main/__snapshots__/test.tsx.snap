// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render component and validate snapshot 1`] = `
.c0 {
  background-color: #fafafb;
  color: #4a4a4a;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  color: #4a4a4a;
  font-size: 2.5rem;
}

.c3 {
  color: #797979;
  font-size: 2rem;
  font-weight: 400;
}

.c1 {
  margin-bottom: 3rem;
  width: min(10rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Um desenvolvedor de frente para uma tela com cÃ³digo."
    class="c1"
    src="/img/react-atom-icon.svg"
  />
  <h1
    class="c2"
  >
    React Boilerplate
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
</main>
`;
